closed/test/jdk/openj9/internal/security/TestProperties.java

```diff
@@ -0,0 +1,181 @@
+ */
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.security.Provider;
+import java.security.Security;
+
+import java.util.stream.Stream;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class TestProperties {
+
+    private static Stream<Arguments> patternMatches_expectedExitValue1() {
+        return Stream.of(
+                // 1 - Test profile - base profile misspell properties.
+                Arguments.of("Test-Profile.Base",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "The property names: RestrictedSecurity.Test-Profile.Base.tls.disabledAlgorithmsWrongTypo "
+                            + "in profile RestrictedSecurity.Test-Profile.Base \\(or a base profile\\) are not recognized"),
+                // 2 - Test profile - extenstion profile misspell properties.
+                Arguments.of("Test-Profile.Extended_1",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "The property names: RestrictedSecurity.Test-Profile.Extended_1.desc.nameWrongTypo, "
+                            + "RestrictedSecurity.Test-Profile.Extended_1.jce.providerWrongTypo in profile "
+                            + "RestrictedSecurity.Test-Profile.Extended_1 \\(or a base profile\\) are not recognized"),
+                // 3 - Test profile - extension profile from another extension profile misspell properties.
+                Arguments.of("Test-Profile.Extended_2",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "The property names: RestrictedSecurity.Test-Profile.Extended_2.jce.providerWrongTypo "
+                            + "in profile RestrictedSecurity.Test-Profile.Extended_2 \\(or a base profile\\) are not recognized"),
+                // 4 - Test profile - profile not exist.
+                Arguments.of("Test-Profile-NotExist.Base",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "Test-Profile-NotExist.Base is not present in the java.security file."),
+                // 5 - Test profile - Multi Default profile.
+                Arguments.of("Test-Profile-MultiDefault",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "Multiple default RestrictedSecurity profiles for Test-Profile-MultiDefault"),
+                // 6 - Test profile - no default profile.
+                Arguments.of("Test-Profile-NoDefault",
+                        System.getProperty("test.src") + "/property-java.security",
+                        "No default RestrictedSecurity profile was found for Test-Profile-NoDefault"),
+                // 7 - Test profile - base profile not exist.
```

Comment:
Missing "does" in "base profile does not exist".

Reply:
Done.